File: LeetCode/0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> largestValues(TreeNode root) {
        Queue<TreeNode> q = new LinkedList<>();
        q.offer(root);
        List<Integer> ans = new ArrayList<>();
        if (root == null) return ans;
        ans.add(root.val);
        while (!q.isEmpty()) {
            int n = q.size();
            boolean flag = false;
            int maxVal = 0;
            while (n-- > 0) {
                TreeNode temp = q.poll();
                if (temp.left != null) {
                    if (flag == false) maxVal = temp.left.val;
                    flag = true;
                    q.offer(temp.left);
                    maxVal = Math.max(maxVal, temp.left.val);
                }
                if (temp.right != null) {
                    if (flag == false) maxVal = temp.right.val;
                    flag = true;
                    q.offer(temp.right);
                    maxVal = Math.max(maxVal, temp.right.val);
                }
            }
            if (flag) ans.add(maxVal);
        }
        return ans;
    }
}

File: LeetCode/0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def largestValues(self, root: Optional[TreeNode]) -> List[int]:
        q = deque()
        q.append(root)
        ans = []
        if not root:
            return ans
        ans.append(root.val)
        while q:
            n = len(q)
            flag = False
            maxVal = 0
            while n:
                n -= 1
                temp = q.popleft()
                if temp.left:
                    if not flag:
                        maxVal = temp.left.val
                    flag = True
                    q.append(temp.left)
                    maxVal = max(maxVal, temp.left.val)
                if temp.right:
                    if not flag:
                        maxVal = temp.right.val
                    flag = True
                    q.append(temp.right)
                    maxVal = max(maxVal, temp.right.val)
            if flag:
                ans.append(maxVal)
        return ans